src/wallet/rpc/spend.cpp

```diff
@@ -221,6 +223,194 @@ static void SetFeeEstimateMode(const CWallet& wallet, CCoinControl& cc, const Un
+ * Valid examples:
+ *   - carol.user._bitcoin-payment.carol.com
+ *   - carol.user._bitcoin-payment.carol.com. (with trailing dot - FQDN)
+ *   - alice._bitcoin-payment.example.org
+ *   - pay.bob._bitcoin-payment.company.co.uk
+ *
+ * Invalid examples:
+ *   - carol.user.carol.com (missing ._bitcoin-payment.)
+ *   - _bitcoin-payment.example.com (missing username part)
+ *   - user@example.com (not a valid DNS name format)
+ */
+bool IsValidBIP353Username(const std::string& username)
+{
+    // Check for empty string
+    if (username.empty()) {
+        return false;
+    }
+
+    // Handle trailing dot (FQDN format)
+    std::string name = username;
+    if (!name.empty() && name.back() == '.') {
+        name.pop_back(); // Remove trailing dot for validation
+
+        // Check if it was just a dot
+        if (name.empty()) {
+            return false;
+        }
+    }
+
+    // Must contain the BIP 353 identifier
+    if (name.find("._bitcoin-payment.") == std::string::npos) {
+        return false;
+    }
+
+    // Basic DNS name validation
+    // - Maximum length of 253 characters (excluding trailing dot)
+    // - Labels separated by dots
+    // - Each label: 1-63 characters, alphanumeric and hyphens
+    // - Labels cannot start or end with hyphens
+    if (name.length() > 253) {
+        return false;
+    }
+
+    // Check for valid characters (alphanumeric, dots, hyphens, underscores)
+    // Use locale-independent character checking
+    if (!std::all_of(name.begin(), name.end(), [](unsigned char c) {
+        return (c >= 'a' && c <= 'z') ||
+               (c >= 'A' && c <= 'Z') ||
+               (c >= '0' && c <= '9') ||
+               c == '.' || c == '-' || c == '_';
+    })) {
+        return false;
+    }
+
+    // Split by dots and validate each label
+    size_t start = 0;
+    size_t dot_pos = name.find('.');
+
+    while (dot_pos != std::string::npos || start < name.length()) {
+        size_t end = (dot_pos != std::string::npos) ? dot_pos : name.length();
+        std::string label = name.substr(start, end - start);
+
+        // Empty label (consecutive dots)
+        if (label.empty()) {
+            return false;
+        }
+
+        // Label too long
+        if (label.length() > 63) {
+            return false;
+        }
+
+        // Label starts or ends with hyphen (unless it's the special _bitcoin-payment label)
+        if (label != "_bitcoin-payment") {
+            if (label.front() == '-' || label.back() == '-') {
+                return false;
+            }
+        }
+
+        if (dot_pos == std::string::npos) {
+            break;
+        }
+
+        start = dot_pos + 1;
+        dot_pos = name.find('.', start);
+    }
+
+    // Ensure it ends with a valid TLD (at least 2 characters after the last dot)
+    size_t last_dot = name.rfind('.');
+    if (last_dot == std::string::npos || name.length() - last_dot < 3) {
+        return false;
+    }
+
+    return true;
+}
+
+struct BIP353Result {
```

Comment:
Maybe return an array of strings, because there might be multiple options we can pay to (regular address, silent payment, some future thing).